DSharpPlus/Entities/Guild/DiscordMember.cs

```diff
@@ -195,6 +196,21 @@ public override string Discriminator
             internal set => this.User.Discriminator = value;
         }
 
+
+
+        /// <summary>
+        /// Gets the member's avatar for the current guild.
+        /// </summary>
+        public string GuildAvatarHash => this._avatarHash ?? this.User.AvatarHash;
+
+        /// <summary>
+        /// Gets the members avatar url for the current guild.
+        /// </summary>
+        public string GuildAvatarUrl
+            => !string.IsNullOrWhiteSpace(this.GuildAvatarHash) ? (this.GuildAvatarHash.StartsWith("a_") ? $"https://cdn.discordapp.com/avatars/{this.Id.ToString(CultureInfo.InvariantCulture)}/{this.GuildAvatarHash}.gif?size=1024" : $"https://cdn.discordapp.com/avatars/{this.Id}/{this.GuildAvatarHash}.png?size=1024") : this.DefaultAvatarUrl;
+
+        private string _avatarHash;
```

Comment:
Should be next to the other non-overrided elements, also, not sure but I think it's missing JsonProperty/JsonIgnore for the different fields/properties you added.

Reply:
Thanks for pointing it out. I actually decided to take a look and it turns out I scuffed the url anwyays, so that needed to be fixed, too. user avatar is `/avatars/:userid/:avatarhash` where as per-server avatar is `/guilds/:guildid/users/:userid/avatars/:avatarhash`